ibazel/fs_watcher_darwin.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build darwin
+
+package main
+
+import (
+	"os"
+
+	"github.com/bazelbuild/bazel-watcher/ibazel/fs_watcher"
+	"github.com/bazelbuild/bazel-watcher/ibazel/fsnotify_watcher"
+	"github.com/bazelbuild/bazel-watcher/ibazel/fsevents_watcher"
+)
+
+func NewWatcher() (fs_watcher.Watcher, error) {
+	flag, ok := os.LookupEnv("IBAZEL_USE_LEGACY_WATCHER") 
+	if ok && flag != "0" {
+		return fsnotify_watcher.NewWatcher()
+	}
+	return fsevents_watcher.NewWatcher()
```

Comment:
Could we wrap the event handler here to add a note to the user that says something like "You're using an experimental filesystem watcher. If you would like to disable that, please set this environment variable" if something goes wrong?

Reply:
Yes, this is a great way for users to be made aware of the change and disable it if needed!